appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/util/ResourceValidator.java

```diff
@@ -833,15 +833,15 @@ private boolean isEmbedded(Application application, String raname) {
      */
     private void validateJNDIRefs(DeploymentContext deploymentContext, Application application, AppResource resource, JNDINamespace namespace) {
         // In case lookup is not present, check if another resource with the same name exists
-        if (!resource.hasLookup() && !namespace.find(resource.getName(), resource.getEnv())) {
+        String jndiName = resource.getJndiName();
+        if (jndiName == null || !resource.hasLookup() && !namespace.find(resource.getName(), resource.getEnv())) {
```

Comment:
I believe the below `throws` also was meant to occur in case `jndiName` is `null`. This is suggested both by the method comment and the parameters to the message passing `null` for the "Lookup:" value (which is another name of the jndiName).